### PR TITLE
Add multi prompt support for beam search

### PR DESCRIPTION
**Pull request by as-suvorov**

### CODE REVIEW

File: text_generation/causal_lm/cpp/group_beam_searcher.hpp [hunk 1]
```diff
@@ -148,42 +148,80 @@ struct Group {
...
                 }
             }
+
+            prompt_id += 1;
+        }
+
+        for (int prompt_id = 0; prompt_id < promts_size; prompt_id++) {
+            const std::vector<int64_t> prompt = parameters.prompts[prompt_id];
+            std::vector<Group>& groups = prompts_groups[prompt_id];
+            auto [prompt_next_tokens, prompt_next_beams] = select_prompt_next_tokens(logits, prompt, groups);
```
[the remark this answers is not in the document]
Comment (as-suvorov) [reply text_generation/causal_lm/cpp/group_beam_searcher.hpp:203]: Good point, attention_mask should be handled differently. I updated implementation, now attention_mask set based on beam_id.

File: text_generation/causal_lm/cpp/beam_search_causal_lm.cpp [hunk 2]
```diff
@@ -22,52 +18,187 @@ std::string detokenize(ov::InferRequest& detokenizer, const std::vector<int64_t>
...
+            std::swap(inputs_data[token_offset], inputs_data[token_offset + pad_tokens_number]);
+            std::swap(attention_mask_data[token_offset], attention_mask_data[token_offset + pad_tokens_number]);
+        }
+    }
+
+    return {input_ids, attention_mask};
+}
+
+std::pair<ov::Tensor, ov::Tensor> tokenize(ov::InferRequest& tokenizer, std::vector<std::string> prompts) {
+    tokenizer.set_input_tensor(ov::Tensor{ov::element::string, {prompts.size()}, prompts.data()});
```
Comment (ilya-lavrenov) [inline text_generation/causal_lm/cpp/beam_search_causal_lm.cpp:57]: side note:
currently, we use batched inference with tokenizers and it creates attention mask for us, which we need to "parse" later. Maybe, alternatively, we could tokenize prompt one by one and it will return us raw (unpadded) data, which we can use more optimally to fill position_ids, etc.

I'm not sure which solution is more optimal, so, let's stick to current one, because it's already implemented.

**Review by ilya-lavrenov — APPROVED** [review]
